enclave/core/sgx/calls.c

```diff
@@ -505,6 +506,118 @@ oe_result_t oe_get_enclave_status()
     return __oe_enclave_status;
 }
 
+/*
+**==============================================================================
+**
+** _exit_enclave()
+**
+** Exit the enclave.
+** Additionally, if a debug enclave, write the exit frame information to host's
+** ecall_context so that the host can stitch the ocall stack.
+**
+** This function is intended to be called by oe_asm_exit (see below).
+** When called, the call stack would look like this:
+**
+**     enclave-function
+**       -> oe_ocall
+**         -> oe_exit_enclave (aliased as __morestack)
+**           -> _exit_enclave
+**
+** For debug enclaves, _exit_enclave reads its caller (oe_exit_enclave/
+** __morestack) information (return address, rbp) and passes it along to the
+** host in the ecall_context.
+**
+** Then it proceeds to exit the enclave by invoking oe_asm_exit.
+** oe_asm_exit invokes eexit instruction which resumes execution in host at the
+** oe_enter function. The host dispatches the ocall via the following sequence:
+**
+**     oe_enter
+**       -> __oe_host_stack_bridge   (Stitches the ocall stack; disappears)
+**         -> __oe_dispatch_ocall
+**           -> invoke ocall function
+**
+** Now that the enclave exit frame is available to the host,
+**__oe_host_stack_bridge temporarily modifies its caller info with the enclave's
```

Comment:
Nit: missing space between `** __oe_host_stack_bridge`

Reply:
Fixed. Also updated the comment to accurately reflect the stitched stack.